### PR TITLE
Add name to IOCell definition

### PR DESCRIPTION
**Pull request by kevindna**
Added variable to i_name in IOCell to contain name of signal to IOBinders

### CODE REVIEW

File: src/main/scala/barstools/iocell/chisel/IOCell.scala [hunk 1]
```diff
@@ -59,7 +59,9 @@ class DigitalInIOCellBundle extends Bundle {
   val ie = Input(Bool())
 }
 
-trait IOCell extends BaseModule
+trait IOCell extends BaseModule {
+  var i_name: String
```
Comment (jerryz123) [inline src/main/scala/barstools/iocell/chisel/IOCell.scala:63]: Can you make this name more descriptive? Also try
```
var iocell_name: Option[String] = None
override def suggestName(s: String): IOCell = {
  iocell_name = Some(s)
  super.suggestName(s)
}
```

If you do it this way, you don't have to set a default in the classes which inherit this trait. 
Also you don't have to do another suggestName in generateIOCell

File: src/main/scala/barstools/iocell/chisel/IOCell.scala [hunk 2]
```diff
@@ -112,7 +118,11 @@ case class GenericIOCellParams() extends IOCellTypeParams {
   def output() = Module(new GenericDigitalOutIOCell)
 }
 
-object IOCell {
+trait IOCellName {
+  var i_name: String
```
Comment (jerryz123) [inline src/main/scala/barstools/iocell/chisel/IOCell.scala:122]: This shouldn't be necessary

File: src/main/scala/barstools/iocell/chisel/IOCell.scala [hunk 3]
```diff
@@ -112,7 +118,11 @@ case class GenericIOCellParams() extends IOCellTypeParams {
...
+}
+
+object IOCell extends IOCellName {
```
Comment (jerryz123) [inline src/main/scala/barstools/iocell/chisel/IOCell.scala:126]: This shouldn't be necessary

File: src/main/scala/barstools/iocell/chisel/IOCell.scala [hunk 4]
```diff
@@ -156,18 +168,25 @@ object IOCell {
     )(coreSignal:   T,
       padSignal:    T
     ): Seq[IOCell] = {
+      print("Suggested names:  " + name + "    ")
```
Comment (jerryz123) [inline src/main/scala/barstools/iocell/chisel/IOCell.scala:171]: Remove this print statement? Or do you think its helpful to have this here?

If you think it is helpful, we should make it more descriptive: 
"Generating IOCell [IOCellType] named [name]"

**Review by jerryz123 — CHANGES_REQUESTED** [review]

File: src/main/scala/barstools/iocell/chisel/IOCell.scala (hunk 4 above)
Comment (kevindna) [reply src/main/scala/barstools/iocell/chisel/IOCell.scala:171]: not helpful. just missed this one. will remove

File: src/main/scala/barstools/iocell/chisel/IOCell.scala (hunk 1 above)
Comment (jerryz123) [reply src/main/scala/barstools/iocell/chisel/IOCell.scala:63]: Yeah. It minimizes the amount of code which has to change, any thing which tries to name an IOCell with suggestName will also set the stored io_cell name with this change

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**